jwst/datamodels/tests/test_fits.py

```diff
@@ -68,3 +69,24 @@ def test_casting():
         total = dm.data.sum()
         dm.data = dm.data + 2
         assert dm.data.sum() > total
+
+
+@pytest.mark.skip(reason='fix required for asdf')
+def test_resave_duplication_bug(tmp_path):
+    """
+    An issue in asdf (https://github.com/asdf-format/asdf/issues/1232)
+    resulted in duplication of data when a model was read from and then
+    written to a fits file.
+    """
+    fn1 = tmp_path / "test1.fits"
+    fn2 = tmp_path / "test2.fits"
+
+    arr = np.zeros((1000, 100), dtype='f4')
+    m = ImageModel(arr)
+    m.save(fn1)
+
+    m2 = jwst.datamodels.open(fn1)
+    m2.save(fn2)
```

Comment:
Use the context manager to make sure it is closed on exit.

```suggestion
with datamodels.open(fn1) as m2:
    m2.save(fn2)
```